libclc/clc/lib/generic/math/clc_mad.inc

```diff
@@ -0,0 +1,4 @@
+_CLC_OVERLOAD _CLC_DEF __CLC_GENTYPE __clc_mad(__CLC_GENTYPE a, __CLC_GENTYPE b,
+                                               __CLC_GENTYPE c) {
+  return a * b + c;
```

Comment:
probably should use the pragma to explicitly allow contract

Reply:
Good idea - done. The default is `ON` so we shouldn't expect any changes from this.